internal/controller/csiaddons/csiaddonsnode_controller.go

```diff
@@ -250,15 +250,20 @@ func parseEndpoint(rawURL string) (string, string, string, error) {
 	}
 
 	// split hostname -> pod.namespace
-	parts := strings.Split(endpoint.Hostname(), ".")
-	podname := parts[0]
-	namespace := ""
-	if len(parts) == 2 {
-		namespace = parts[1]
-	} else if len(parts) > 2 {
-		return "", "", "", fmt.Errorf("hostname %q is not in <pod>.<namespace> format", endpoint.Hostname())
+	hostName := endpoint.Hostname()
+	lastIndex := strings.LastIndex(hostName, ".")
+
+	if lastIndex == 0 {
+		return "", "", "", fmt.Errorf("hostname %q is not in <pod>.<namespace> format", hostName)
+	}
+
+	if lastIndex == -1 {
+		return "", hostName, endpoint.Port(), nil
```

Comment:
Let's return an error if we dont find the namespace as well, because we need that and from the caller we are erroring out if the namespace is not set, i don't see a code where we are choosing the default namespace if the return namespace value is empty from here.

Reply:
done